mbed_greentea/mbed_greentea_cli.py

```diff
@@ -661,6 +739,69 @@ def get_parallel_value(value):
     if not test_spec:
         return ret
 
+    available_test_cases = test_spec.get_test_names_by_test_cases()
+    filtered_test_cases = []
+    # Populate list of the test cases wanted from the test plan
+    # All test cases have to have unique names
+    if opts.test_plan:
+        test_plan_json = None
+        try:
+            with open(opts.test_plan, "r") as f:
+                test_plan_json = json.load(f)
+        except Exception as e:
+            gt_logger.gt_log_err("loading Test Plan('%s')"% opts.test_plan + str(e))
+
+        if test_plan_json:
+            test_plan = test_plan_json['testplan']
+            gt_logger.gt_log("using Test Plan '%s'"% gt_logger.gt_bright(test_plan['name']))
+            
+            for test_case in test_plan['testcases']:
+                filtered_test_cases.append(test_case['name'])
+
+        # Check for conflicting test cases and fail if any are found
+        conflicting = {}
+        for build in test_spec.get_test_builds():
+            conflicting[build.get_name()] = []
+            all_test_cases = []
+            for test_case in build.get_test_cases():
+                if test_case in all_test_cases:
+                    conflicting[build.get_name()].append(test_case)
+                all_test_cases.append(test_case)
+        if all(conflicting.values()):
+            for build, test_cases in conflicting.iteritems():
+                gt_logger.gt_log_err("conflicting test cases (name not unique) in build '%s'"% build)
+                for tc in test_cases:
+                    gt_logger.gt_log_tab("test case '%s'"% gt_logger.gt_bright(tc))
+            return(-1)
+
+    if opts.list_test_plan:
+        all_found = []
+        implemented = []
+        not_implemented = []
+        conflicting = []
+        for test_case in filtered_test_cases:
+            if test_case in all_found:
+                conflicting.append(test_case)
+            elif test_case not in available_test_cases:
+                not_implemented.append(test_case)
+            else:
+                implemented.append(test_case)
+            all_found.append(test_case)
```

Comment:
It seems like the above logic could be consolidated with some of the checks above? The `conflicting`, `all_found`, etc.